applications/DEMApplication/custom_strategies/strategies/explicit_solver_strategy.cpp

```diff
@@ -1618,56 +1604,40 @@ namespace Kratos {
+
         KRATOS_TRY
+
         //CONTACT MODEL PART
         ElementsArrayType& pContactElements = GetAllElements(*mpContact_model_part);
-        std::vector<unsigned int> contact_element_partition;
-        OpenMPUtils::CreatePartition(mNumberOfThreads, pContactElements.size(), contact_element_partition);
         const ProcessInfo& r_process_info = GetModelPart().GetProcessInfo();
 
-        #pragma omp parallel for
-        for (int k = 0; k < mNumberOfThreads; k++) {
-            ElementsArrayType::iterator it_contact_begin = pContactElements.ptr_begin() + contact_element_partition[k];
-            ElementsArrayType::iterator it_contact_end = pContactElements.ptr_begin() + contact_element_partition[k + 1];
-
-            for (ElementsArrayType::iterator it_contact = it_contact_begin; it_contact != it_contact_end; ++it_contact) {
-                (it_contact)->Initialize(r_process_info);
-            } //loop over CONTACT ELEMENTS
-        }// loop threads OpenMP
+        block_for_each(pContactElements, [&](ModelPart::ElementType& rContactElement) {
+            rContactElement.Initialize(r_process_info);
+        });
 
         KRATOS_CATCH("")
     }
 
     void ExplicitSolverStrategy::PrepareContactElementsForPrinting() {
 
         ElementsArrayType& pContactElements = GetAllElements(*mpContact_model_part);
-        std::vector<unsigned int> contact_element_partition;
-
-        OpenMPUtils::CreatePartition(mNumberOfThreads, pContactElements.size(), contact_element_partition);
 
         #pragma omp parallel for
-        for (int k = 0; k < mNumberOfThreads; k++) {
-            ElementsArrayType::iterator it_contact_begin = pContactElements.ptr_begin() + contact_element_partition[k];
-            ElementsArrayType::iterator it_contact_end = pContactElements.ptr_begin() + contact_element_partition[k + 1];
-
-            for (ElementsArrayType::iterator it_contact = it_contact_begin; it_contact != it_contact_end; ++it_contact) {
-                Element* raw_p_contact_element = &(*it_contact);
-                ParticleContactElement* p_bond = dynamic_cast<ParticleContactElement*> (raw_p_contact_element);
-                p_bond->PrepareForPrinting();
-            } //loop over CONTACT ELEMENTS
-        }// loop threads OpenMP
-        //Important TODO: renumber all id's to avoid repetition across partitions
+        for (int k = 0; k < (int) pContactElements.size(); k++) {
+            ElementsArrayType::iterator it_contact = pContactElements.ptr_begin() + k;
+            Element* raw_p_contact_element = &(*it_contact);
+            ParticleContactElement* p_bond = dynamic_cast<ParticleContactElement*> (raw_p_contact_element);
+            p_bond->PrepareForPrinting();
+        }
     } //PrepareContactElementsForPrinting
 
     void ExplicitSolverStrategy::ComputeNewRigidFaceNeighboursHistoricalData() {
         KRATOS_TRY
         const int number_of_particles = (int) mListOfSphericParticles.size();
-
         #pragma omp parallel for
```

Comment:
same

Reply:
not sure about this one

Reply:
here you need to use the `IndexPartition`

Reply:
Oh I see, 
thx!